integration-tests/tests/client/process_blocks.rs

```diff
@@ -3938,4 +3942,181 @@ mod contract_precompilation_tests {
         // Check that contract is not cached for client 1 because of late state sync.
         assert!(caches[1].get(&contract_key.0).unwrap().is_none());
     }
+
+    #[test]
+    fn test_block_producers_with_sampling() {
```

Comment:
Is it intended that this function is inside `contract_precompilation_tests`? Feels unrelated to contract precompilation.